### PR TITLE
feat: add initial v3 converter support

### LINKED ISSUE #110: Support converting to 3.0.0
It's a long term feature request. 

When we will have the `3.0.0` specification stabilized we should start writing functionality for converting versions `2.X.X` to `3.0.0`.

One of the problems I see are the external sources (local files and http links) which we are not converting at the moment but we should (see issue https://github.com/asyncapi/converter-js/issues/90). In order not to cause problems for people and increase DX we should figure out a way to convert also external files in which messages (and other reusable components) can be defined.

### Scope
- [x] #188
- [ ] https://github.com/asyncapi/converter-js/issues/90
- [x] https://github.com/asyncapi/converter-js/issues/106
- [x] https://github.com/asyncapi/converter-js/issues/107

### LINKED ISSUE #188: Convert `2.X.X` to `3.0.0`
Related to #110

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR is an extension to what @magicmatatjahu did in https://github.com/asyncapi/converter-js/pull/163

TODO:

- [x] Add support for https://github.com/asyncapi/spec/issues/622

TODO after merge:
- Add parameter conversion

**Related issue(s)**
Part of #110 
Fixes #188

### CODE REVIEW

**Comment by jonaslagoni** [comment]
I cant figure out whether we need to add a warning if traits + target object contains the same property (i.e. the breaking behavior of https://github.com/asyncapi/spec/pull/907).

**Comment by jonaslagoni** [comment]
Should we do binding conversion as well? 🤔

**Comment by jonaslagoni** [comment]
Still missing adding change for: https://github.com/asyncapi/spec/pull/935

**Comment by jonaslagoni** [comment]
Even though this PR is not the final PR for v3 conversion, I will say this is a good enough state that we can release it as it is right now. - And directly to master, I don't see why not, although it's an experimental feature 😄 

You do have to explicitly define that you want to convert to v3.

File: src/third-version.ts [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+  if (securityScheme.flows) {
+    flowKinds.forEach(flow => {
+      const flowScheme = securityScheme.flows[flow];
+      if (flowScheme && flowScheme.scopes) {
+        flowScheme.availableScopes = flowScheme.scopes;
+        delete flowScheme.scopes;
+      }
+    });
+  }
+  return securityScheme;
```
Comment (jonaslagoni) [inline src/third-version.ts:387]: @magicmatatjahu where does this change originate from? I cant find anything in the spec changes that suggest this change?

**Review by jonaslagoni — COMMENTED** [review]
Currently known shortcomings:
- Parameters are not converted

File: src/third-version.ts (hunk 1 above)
Comment (magicmatatjahu) [reply src/third-version.ts:387]: https://github.com/asyncapi/spec/blob/next-major-spec/spec/asyncapi.md#fixed-fields-27 It was introduced by Fran

Comment (jonaslagoni) [reply src/third-version.ts:387]: Thanks!

File: README.md [hunk 2]
```diff
@@ -91,6 +92,74 @@ try {
 }
 ```
 
+## Conversion 2.x.x to 3.x.x
+> NOTE: this feature is still WIP, and is until the final release of 3.0.0
```
Comment (magicmatatjahu) [inline README.md:96]:
```suggestion
## Conversion 2.x.x to 3.x.x

> **NOTE**: This feature is still WIP, and is until the final release of `3.0.0`.
```

File: README.md [hunk 3]
```diff
@@ -91,6 +92,74 @@ try {
...
+      action: receive
+      ...
+    userSignUpEvent:
+      action: send
+      ...
+  ```
+
+- Security requirements that use scopes are defined in the appropriate places inline, the rest as a reference to the `components.securitySchemes` objects.
+- If servers are defined at the channel level, they are converted as references to the corresponding objects defined in the `servers` field.
+- Channels and servers defined in components are also converted (unless configured - see [examples](#in-js)).
```
Comment (magicmatatjahu) [inline README.md:161]: You should add example in JS with options to skip `channels` and `servers` conversion.

**Review by magicmatatjahu — DISMISSED** [review]
LGTM! I wrote 90% of this code myself so I recognize that it works 😄 Only I noticed some bugs in the Readme.

**Review by fmvilas — APPROVED** [review]
Just the minor issues that Maciej pointed out. Other than that, 🐑 🇮🇹

**Comment by derberg** [comment]
I see 2 reviewers there already, so no need for me to jump in

**Comment by jonaslagoni** [comment]
Just a reminder @fmvilas @magicmatatjahu I am targeting this PR to `master` in case you missed that information - and using the explicit `3.0.0` conversion target as a feature flag, avoiding to work on a feature branch 😄

File: README.md (hunk 3 above)
Comment (magicmatatjahu) [reply README.md:162]: So maybe remove info about examples from this list item? And add after release 3.0.0.

**Review by fmvilas — APPROVED** [review]

**Comment by derberg** [comment]
code smells 😄

File: README.md (hunk 3 above)
Comment (jonaslagoni) [reply README.md:162]: Yep, removed 👍

**Comment by jonaslagoni** [comment]
I refactored everything that seemed problematic, and nothing extra has been added 🙂 

Need a final review @derberg @fmvilas @magicmatatjahu

**Comment by jonaslagoni** [comment]
Anytime now @fmvilas @derberg @magicmatatjahu 😆

**Review by derberg — APPROVED** [review]
nice to see it coming ❤️ . All good from my side but let's have approval from folks that also did a code review initially

**Comment by derberg** [comment]
@magicmatatjahu any time? or busy "working" from Georgia? 😛

**Review by fmvilas — APPROVED** [review]
:shipit:

**Review by magicmatatjahu — APPROVED** [review]
LGTM!

**Review by magicmatatjahu — APPROVED** [review]

**Comment by jonaslagoni** [comment]
Thanks folks 🥳

**Pull request merged**